packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviderTests.cs

```diff
@@ -0,0 +1,405 @@
+            foreach (var inputParameter in inputParameters)
+            {
+                Assert.IsTrue(fields.Any(f => f.Name == "_" + inputParameter.Name));
+            }
+            Assert.IsTrue(fields.Any(f => f == clientProvider.PipelineField));
+
+            // validate the endpoint field
+            if (inputParameters.Any(p => p.IsEndpoint))
+            {
+                var endpointField = fields.FirstOrDefault(f => f.Name == "_endpoint");
+                Assert.IsNotNull(endpointField);
+                Assert.AreEqual(new CSharpType(typeof(Uri)), endpointField?.Type);
+            }
+        }
+
+        [TestCaseSource(nameof(BuildConstructorsTestCases))]
+        public void TestBuildConstructors_PrimaryConstructor(List<InputParameter> inputParameters, bool containsEndpointParam)
+        {
+            var mockClient = new InputClient("TestClient", "TestClient description", [], inputParameters, null);
+            var clientProvider = new ClientProvider(mockClient);
+
+            Assert.IsNotNull(clientProvider);
+
+            var constructors = clientProvider.Constructors;
+            Assert.AreEqual(3, constructors.Count);
+
+            var primaryPublicConstructor = constructors.FirstOrDefault(
+                c => c.Signature?.Initializer == null && c.Signature?.Modifiers == MethodSignatureModifiers.Public);
+            ValidatePrimaryConstructor(primaryPublicConstructor, inputParameters, containsEndpointParam);
+        }
+
+        [TestCaseSource(nameof(BuildConstructorsTestCases))]
+        public void TestBuildConstructors_SecondaryConstructor(List<InputParameter> inputParameters, bool containsEndpointParam)
+        {
+            var mockClient = new InputClient("TestClient", "TestClient description", [], inputParameters, null);
+            var clientProvider = new ClientProvider(mockClient);
+
+            Assert.IsNotNull(clientProvider);
+
+            var constructors = clientProvider.Constructors;
+
+            Assert.AreEqual(3, constructors.Count);
+            var primaryPublicConstructor = constructors.FirstOrDefault(
+                c => c.Signature?.Initializer == null && c.Signature?.Modifiers == MethodSignatureModifiers.Public);
+
+            Assert.IsNotNull(primaryPublicConstructor);
+
+            var secondaryPublicConstructor = constructors.FirstOrDefault(
+                c => c.Signature?.Initializer != null && c.Signature?.Modifiers == MethodSignatureModifiers.Public);
+            ValidateSecondaryConstructor(primaryPublicConstructor, secondaryPublicConstructor, inputParameters, containsEndpointParam);
+        }
+
+        [TestCaseSource(nameof(BuildConstructorsTestCases))]
+        public void TestBuildConstructors_SubClientConstructors(List<InputParameter> inputParameters, bool containsEndpointParam)
+        {
+            var mockClient = new InputClient("TestClient", "TestClient description", [], inputParameters, "parent");
```

Comment:
nit: I would avoid the word `mock` here as its simply a client we created we aren't mocking anything.